apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts

```diff
@@ -75,32 +108,48 @@ export class DotEditLayoutComponent implements OnInit {
             this.dotMessageService.get('dot.common.message.saving')
         );
 
+        
         this.dotPageLayoutService
-            .save(this.pageState.page.identifier, {
-                ...value,
-                // To save a layout and no a template the title should be null
-                title: null
-            })
+            // To save a layout and no a template the title should be null
+            .save(this.pageState.page.identifier, { ...value, title: null })
             .pipe(take(1))
             .subscribe(
-                (updatedPage: DotPageRender) => {
-                    const mappedContainers = this.getRemappedContainers(updatedPage.containers);
-                    this.templateContainersCacheService.set(mappedContainers);
-
-                    this.dotGlobalMessageService.success(
-                        this.dotMessageService.get('dot.common.message.saved')
-                    );
-                    this.pageState = updatedPage;
-                },
-                (err: ResponseView) => {
-                    this.dotGlobalMessageService.error(err.response.statusText);
-                    this.dotHttpErrorManagerService.handle( new HttpErrorResponse(err.response) ).subscribe(() => {
-                        this.dotEditLayoutService.changeDesactivateState(true);
-                    });
-                }
+                (updatedPage: DotPageRender) => this.onSaveTemplate(updatedPage),
+                (err: ResponseView) => this.onErrorSavingTemplate(err),
+                () => (this.didTemplateChanged = false)
             );
     }
 
+    /**
+     *  Handle next template value;
+     *
+     * @param {DotLayout} value
+     * @memberof DotEditLayoutComponent
+     */
+    nextUpdateTemplate(value: DotLayout) {
+        this.didTemplateChanged = true;
+        this.updateTemplate.next(value);
```

Comment:
Why this and not just call a method?

Reply:
In this component, we are not using the store. Using a `subject` allows me to set a `debounceTime` to the service we are calling here.

[see observable](https://github.com/dotCMS/core-web/pull/1865/files/a840470c3243eba3fbf88cf6c5d2b06160f26db9#diff-cf3e032b3ce4047a8eda1e0e52d12bb19770f142b3697639a62ce4479e6dd6a8R58-R80)